confirm.h

```diff
@@ -27,8 +29,19 @@ struct confirm_query {/*{{{*/
 		};
 	};
 
-	uint16_t icmpid; /* icmpid == 0 fixes reverse flow id =revflow */
-			 /* uint16_t icmpseq used to identify probes */
+	union {
+		struct {
+			uint16_t src_port;
+			uint16_t dst_port;
+			uint32_t ack_number;
+			uint8_t  control_flags;
+			uint16_t window;
+			uint16_t urgent_pointer;
+		} tcp;
+
+		uint16_t icmpid; /* icmpid == 0 fixes reverse flow id =revflow */
```

Comment:
Yeah, we should probably use field names for ICMP as well. Otherwise these things will get confusing. Maybe changing to `probe_id` like you suggested is the way to go.